### ISSUE TITLE
Cannot determine default entity. CRM_Findpayment_Form_Search should implement getDefaultEntity()

### ISSUE THREAD

**Issue by zkrebs** [JMAConsulting/biz.jmaconsulting.findpayments#5]
The extension was working on an earlier version of CiviCRM 5.11, then we updated to 5.16.3 and the Find Payments menu item now throws error:

Cannot determine default entity. CRM_Findpayment_Form_Search should implement getDefaultEntity().

**Comment by JoeMurray** [comment]
@monishdeb pls look into this.

**Comment by zkrebs** [comment]
Is there any update on this or a way to support the resolution? We actually depend on this for a small part of our accounting reconciliation.

**Comment by zkrebs** [comment]
I have no idea what I'm doing, but I googled and found getDefaultEntity() example here: https://lab.civicrm.org/dev/core/blob/267fb11ab759e9665bab90d61f873a60637b1dee/CRM/Pledge/Form/Payment.php

I then edited /biz.jmaconsulting.findpayments-master/CRM/Findpayment/Form/Search.php and added the following on Line 5


```
/**
   * Explicitly declare the entity api name.
   */
  public function getDefaultEntity() {
    return 'Contribution';
  }

```

And the extension now loads without an error and seems to work.

**Comment by monishdeb** [comment]
@zkrebs Sorry for the delay. I am not able to replicate this issue. I have tested it in latest CiviCRM version 5.18.0 and on master (5.19.alpha1) but got no error:
<img width="909" alt="Screen Shot 2019-10-05 at 9 35 45 AM" src="https://user-images.githubusercontent.com/3735621/66249609-ca122780-e753-11e9-880d-ff7df511fda5.png">


The reason is, I am always passing the 'entity' attribute with value 'FinancialTrxn' in [here](https://github.com/JMAConsulting/biz.jmaconsulting.findpayments/blob/master/CRM/Findpayment/BAO/Query.php#L150). When the addField() fn (which is responsible to add filter fields in the search form) wasn't able to find the entity attribute, it looks out for the default entity provided by search form and i.e. returned by ```getDefaultEntity()```.  Is there any customization made in your extension?

**Comment by zkrebs** [comment]
I am still on 5.16.3 - I suppose that could be the issue. 

I made no customizations to the extension whatsoever and received the error. 3 days ago I uninstalled it, deleted it, and then cloned the master branch. I installed that, and still received the error. My "fix" in my above post seems to have resolved it for our purposes. 

My next step will be to upgrade CiviCRM and reinstall without any changes and see if it works.
